### PR TITLE
Add CLIMATE-FEVER dataset

### PR DESCRIPTION
**Pull request by tdiggelm**
As suggested by @SBrandeis , fresh PR that adds CLIMATE-FEVER. Replaces PR #1579.

---

A dataset adopting the FEVER methodology that consists of 1,535 real-world claims regarding climate-change collected on the internet. Each claim is accompanied by five manually annotated evidence sentences retrieved from the English Wikipedia that support, refute or do not give enough information to validate the claim totalling in 7,675 claim-evidence pairs. The dataset features challenging claims that relate multiple facets and disputed cases of claims where both supporting and refuting evidence are present.

More information can be found at:

* Homepage: http://climatefever.ai
* Paper: https://arxiv.org/abs/2012.00614

### CODE REVIEW

File: datasets/climate_fever/README.md [hunk 1]
```diff
@@ -0,0 +1,235 @@
+---
+annotations_creators:
+- crowdsourced
+- expert-generated
+language_creators:
+- found
+languages:
+- en
+licenses: []
```
Comment (lhoestq) [inline datasets/climate_fever/README.md:9]:
```suggestion
licenses:
- unknown
```

File: datasets/climate_fever/README.md [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+multilinguality:
+- monolingual
+size_categories:
+- 1K<n<10K
+source_datasets: []
```
Comment (lhoestq) [inline datasets/climate_fever/README.md:14]:
```suggestion
source_datasets:
- extended|wikipedia
- original
```

File: datasets/climate_fever/README.md [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+- `claim_id`: a `string` feature, unique claim identifier.
+- `claim`: a `string` feature, claim text.
+- `claim_label`: a `string` feature, overall label assigned to claim (based on evidence majority vote).
+- `evidence_id`: a `string` feature,  unique evidence identifier.
+- `evidence_label`: a `string` feature, micro-verdict label.
+- `article`: a `string` feature, title of source article (Wikipedia page).
+- `evidence`: a `string` feature, evidence sentence.
+- `entropy`: a `float32` feature, entropy reflecting uncertainty of `evidence_label`.
+- `votes`: a `list` of `string` features, corresponding to individual votes.
```
Comment (lhoestq) [inline datasets/climate_fever/README.md:164]:
```suggestion
- `evidences`: a list of evidences with fields:
  - `evidence_id`: a `string` feature,  unique evidence identifier.
  - `evidence_label`: a `string` feature, micro-verdict label.
  - `article`: a `string` feature, title of source article (Wikipedia page).
  - `evidence`: a `string` feature, evidence sentence.
  - `entropy`: a `float32` feature, entropy reflecting uncertainty of `evidence_label`.
  - `votes`: a `list` of `string` features, corresponding to individual votes.
```

File: datasets/climate_fever/climate_fever.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+    VERSION = datasets.Version(_VERSION)
+
+    def _info(self):
+
+        features = datasets.Features(
+            {
+                "claim_id": datasets.Value("string"),
+                "claim": datasets.Value("string"),
+                "claim_label": datasets.Value("string"),
```
Comment (lhoestq) [inline datasets/climate_fever/climate_fever.py:66]: Can you use the ClassLabel feature type here ?
For example
```
"claim_label": datasets.ClassLabel(names=["REFUTES", "SUPPORTS", "NOT_ENOUGH_INFO"])
```

you'll need to regenerate the dataset_infos.json file after this change

File: datasets/climate_fever/climate_fever.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+                "evidences": [
+                    {
+                        "evidence_id": datasets.Value("string"),
+                        "evidence_label": datasets.Value("string"),
```
Comment (lhoestq) [inline datasets/climate_fever/climate_fever.py:70]: same here

**Review by lhoestq — COMMENTED** [review]
Really cool thanks !

I just left comments suggesting the use of the ClassLabel feature type :)

**Comment by tdiggelm** [comment]
Thank you @lhoestq for your comments! 😄 I added your suggested changes, ran the tests and regenerated `dataset_infos.json` and `dummy_data`.

File: datasets/climate_fever/README.md [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+- `claim_id`: a `string` feature, unique claim identifier.
+- `claim`: a `string` feature, claim text.
+- `claim_label`: a `string` feature, overall label assigned to claim (based on evidence majority vote).
+- `evidences`: a list of evidences with fields:
+ - `evidence_id`: a `string` feature,  unique evidence identifier.
+ - `evidence_label`: a `string` feature, micro-verdict label.
+ - `article`: a `string` feature, title of source article (Wikipedia page).
+ - `evidence`: a `string` feature, evidence sentence.
+ - `entropy`: a `float32` feature, entropy reflecting uncertainty of `evidence_label`.
+ - `votes`: a `list` of `string` features, corresponding to individual votes.
```
Comment (lhoestq) [inline datasets/climate_fever/README.md:168]:
```suggestion
- `claim_id`: a `string` feature, unique claim identifier.
- `claim`: a `string` feature, claim text.
- `claim_label`: a `int` feature, overall label assigned to claim (based on evidence majority vote). The label correspond to 0: "refutes", 1: "supports" and 2: "not enough info"
- `evidences`: a list of evidences with fields:
 - `evidence_id`: a `string` feature,  unique evidence identifier.
 - `evidence_label`: a `int` feature, micro-verdict label. The label correspond to 0: "refutes", 1: "supports" and 2: "not enough info"
 - `article`: a `string` feature, title of source article (Wikipedia page).
 - `evidence`: a `string` feature, evidence sentence.
 - `entropy`: a `float32` feature, entropy reflecting uncertainty of `evidence_label`.
 - `votes`: a `list` of `string` features, corresponding to individual votes.
```

File: datasets/climate_fever/README.md [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+      "SUPPORTS",
+      "SUPPORTS",
+      null,
+      null,
+      null
+     ]
+    },
+    {
+     "evidence_id": "Polar bear:1328",
+     "evidence_label": "NOT_ENOUGH_INFO",
```
Comment (lhoestq) [inline datasets/climate_fever/README.md:141]:
```suggestion
     "evidence_label": 2,  # "NOT_ENOUGH_INFO"
```

File: datasets/climate_fever/README.md [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+      "SUPPORTS",
+      "NOT_ENOUGH_INFO",
+      null,
+      null,
+      null
+     ]
+    },
+    {
+     "evidence_id": "Habitat destruction:61",
+     "evidence_label": "SUPPORTS",
```
Comment (lhoestq) [inline datasets/climate_fever/README.md:127]:
```suggestion
     "evidence_label": 1,  # "SUPPORTS"
```

File: datasets/climate_fever/README.md [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+      "SUPPORTS",
+      "NOT_ENOUGH_INFO",
+      null,
+      null,
+      null
+     ]
+    },
+    {
+     "evidence_id": "Global warming:14",
+     "evidence_label": "SUPPORTS",
```
Comment (lhoestq) [inline datasets/climate_fever/README.md:99]:
```suggestion
     "evidence_label": 1,  # "SUPPORTS"
```

File: datasets/climate_fever/README.md [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+
+## Dataset Structure
+
+### Data Instances
+
+```json
+{
+  "claim_id": "0",
+  "claim": "Global warming is driving polar bears toward extinction",
+  "claim_label": "SUPPORTS",
```
Comment (lhoestq) [inline datasets/climate_fever/README.md:81]:
```suggestion
  "claim_label": 1,  # "SUPPORTS"
```

File: datasets/climate_fever/README.md [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+  "evidences": [
+    {
+     "evidence_id": "Extinction risk from global warming:170",
+     "evidence_label": "NOT_ENOUGH_INFO",
```
Comment (lhoestq) [inline datasets/climate_fever/README.md:85]:
```suggestion
     "evidence_label": 2,  # "NOT_ENOUGH_INFO"
```

**Review by lhoestq — APPROVED** [review]
Thank you !
Looks all good now :)

File: datasets/climate_fever/README.md [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+      "SUPPORTS",
+      "SUPPORTS",
+      null,
+      null,
+      null
+     ]
+    },
+    {
+     "evidence_id": "Global warming:178",
+     "evidence_label": "NOT_ENOUGH_INFO",
```
Comment (lhoestq) [inline datasets/climate_fever/README.md:113]:
```suggestion
     "evidence_label": 2,  # "NOT_ENOUGH_INFO"
```

File: datasets/climate_fever/README.md [hunk 13]
```diff
@@ -0,0 +1,239 @@
...
+
+### Languages
+
+The text in the dataset is in English, as found in real-world claims about climate-change on the Internet. The associated BCP-47 code is `en`.
+
+## Dataset Structure
+
+### Data Instances
+
+```json
```
Comment (lhoestq) [inline datasets/climate_fever/README.md:77]:
````suggestion
```
````

**Pull request merged**